lib/data/sender.js

```diff
@@ -1,70 +1,67 @@
 'use strict';
 
-var inherits = require('util').inherits;
-var DataTrackTransceiver = require('./transceiver');
-var makeUUID = require('../util').makeUUID;
+const DataTrackTransceiver = require('./transceiver');
+const makeUUID = require('../util').makeUUID;
 
 /**
- * Construct a {@link DataTrackSender}.
- * @class
- * @classdesc A {@link DataTrackSender} represents a
- *   {@link DataTrackTransceiver} over which data can be sent. Internally, it uses a
- *   collection of RTCDataChannels to send data.
+ * A {@link DataTrackSender} represents a {@link DataTrackTransceiver} over
+ * which data can be sent. Internally, it uses a collection of RTCDataChannels
+ * to send data.
  * @extends DataTrackTransceiver
- * @param {?number} maxPacketLifeTime
- * @param {?number} maxRetransmits
- * @param {boolean} ordered
  */
-function DataTrackSender(maxPacketLifeTime, maxRetransmtis, ordered) {
-  if (!(this instanceof DataTrackSender)) {
-    return new DataTrackSender(maxPacketLifeTime, maxRetransmtis, ordered);
+class DataTrackSender extends DataTrackTransceiver {
+  /**
+   * Construct a {@link DataTrackSender}.
+   * @param {?number} maxPacketLifeTime
+   * @param {?number} maxRetransmits
+   * @param {boolean} ordered
+   */
+  constructor(maxPacketLifeTime, maxRetransmtis, ordered) {
+    const id = makeUUID();
+    super(id, maxPacketLifeTime, maxRetransmtis, ordered);
+    Object.defineProperties(this, {
```

Comment:
@markandrus Can we declare this as a `private readonly` class member if es6 allows it? Then I don't think we need to do this manually.

Reply:
`private` and `readonly` are TypeScript features. We can introduce these when we move to TypeScript 👍